worlds/ff4fe/__init__.py

```diff
@@ -0,0 +1,488 @@
+import json
+import os
+import threading
+import typing
+from typing import Mapping, Any
+
+import Utils
+import settings
+from BaseClasses import Region, ItemClassification, MultiWorld, Tutorial
+from worlds.AutoWorld import World, WebWorld
+from worlds.generic.Rules import set_rule, add_rule, add_item_rule
+from . import events, items, locations
+from . import rules as FERules
+from .Client import FF4FEClient
+from .FreeEnterpriseForAP.FreeEnt.cmd_make import MakeCommand
+from .itempool import create_itempool
+from .items import FF4FEItem, all_items, ItemData
+from .locations import FF4FELocation, all_locations, LocationData
+from .options import FF4FEOptions, ff4fe_option_groups
+from . import topology, flags
+from .rom import FF4FEProcedurePatch
+
+
+class FF4FEWebWorld(WebWorld):
+    theme = "grassFlowers"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Final Fantasy IV: Free Enterprise randomizer connected to an Archipelago Multiworld.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Rosalie"]
+    )
+
+    tutorials = [setup_en]
+
+    option_groups = ff4fe_option_groups
+
+
+class FF4FESettings(settings.Group):
+    class RomFile(settings.UserFilePath):
+        """File name of the FF4 USA 1.1 US rom"""
+
+        copy_to = "Final Fantasy II (USA) (Rev A).sfc"
+        description = "FFII SNES 1.1 ROM File"
+        md5s = ["27D02A4F03E172E029C9B82AC3DB79F7"]
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+    rom_start: bool = True
+
+
+class FF4FEWorld(World):
+    """Final Fantasy IV: Free Enterprise is an open world randomizer for the classic SNES RPG. Explore the world,
+    find the Crystal, and defeat Zeromus on the moon. Adapted from the open source release of FE 4.6.0."""
+    game = "Final Fantasy IV Free Enterprise"
+    options_dataclass = FF4FEOptions
+    options: FF4FEOptions
+    settings: typing.ClassVar[FF4FESettings]
+
+    web = FF4FEWebWorld()
+    topology_present = True
```

Comment:
I definitely maybe thought I had a good reason at one point, but mostly it let me confirm I set the regions and locations up correctly, so it can be removed, yeah. Will do so.